compiler/rustc_error_messages/locales/en-US/lint.ftl

```diff
@@ -84,6 +84,11 @@ lint_ty_qualified = usage of qualified `ty::{$ty}`
 lint_lintpass_by_hand = implementing `LintPass` by hand
     .help = try using `declare_lint_pass!` or `impl_lint_pass!` instead
 
+lint_ty_eq_operator = using the `==` operator on `ty::Ty`
+    .note = this does probably not what you want as it does not handle inference variables and more
+    .note2 = it's also not recommended to use it for diagnostics
+    .help = for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#someone-pr-this-pls
```

Comment:
https://rustc-dev-guide.rust-lang.org/ty.html#allocating-and-working-with-types has a block like this:
> 

    N.B. Because types are interned, it is possible to compare them for equality efficiently using == – however, this is almost never what you want to do unless you happen to be hashing and looking for duplicates. This is because often in Rust there are multiple ways to represent the same type, particularly once inference is involved. If you are going to be testing for type equality, you probably need to start looking into the inference code to do it right.

Reply:
I have made a PR to extend this section a little more rust-lang/rustc-dev-guide#1570